extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -412,6 +419,12 @@ public boolean publishSegments(Set<DataSegment> segments, Object commitMetadata)
             throw new ISE("WTF?! Driver attempted to publish invalid metadata[%s].", commitMetadata);
           }
 
+          // Upgrade TaskLocks for all segments
+          for (DataSegment segment: segments) {
+            toolbox.getTaskActionClient().submit(new SetLockCriticalStateAction(segment.getInterval(),
+                                                                                TaskLockCriticalState.UPGRADE));
```

Comment:
I meant exact same interval, I search for a `TaskLockPosse` for this task having the same interval. Doesn't kafkaIndexTask will acquire lock as it announces the segments ?